atomic_reactor/plugins/post_tag_and_push.py

```diff
@@ -134,18 +139,40 @@ def run(self):
 
                 registry_image = image.copy()
                 registry_image.registry = registry
-                if self.need_skopeo_push():
-                    self.push_with_skopeo(registry_image, insecure, docker_push_secret)
-                else:
-                    self.tasker.tag_and_push_image(self.workflow.builder.image_id,
-                                                   registry_image, insecure=insecure,
-                                                   force=True, dockercfg=docker_push_secret)
-                    defer_removal(self.workflow, registry_image)
+                max_retries = DOCKER_PUSH_MAX_RETRIES
+
+                expect_v2s2 = False
+                for registry in self.registries:
+                    media_types = self.registries[registry].get('expected_media_types', [])
+                    if MEDIA_TYPE_DOCKER_V2_SCHEMA2 in media_types:
+                        expect_v2s2 = True
+
+                if not (self.group or expect_v2s2):
+                    max_retries = 0
+
+                for retry in range(max_retries + 1):
+                    if self.need_skopeo_push():
+                        self.push_with_skopeo(registry_image, insecure, docker_push_secret)
+                    else:
+                        self.tasker.tag_and_push_image(self.workflow.builder.image_id,
+                                                       registry_image, insecure=insecure,
+                                                       force=True, dockercfg=docker_push_secret)
+
+                    digests = get_manifest_digests(registry_image, registry,
+                                                   insecure, docker_push_secret)
+
+                    if (not (digests.v2 or digests.oci) and (retry != max_retries)):
```

Comment:
IMO it is safer to use `retry < max_retries` in case of accidental increment

Reply:
sure